src/page/Page.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'redux-bundler-react'
+import { Helmet } from 'react-helmet'
+import isIPFS from 'is-ipfs'
+
+// Constants
+import PAGES from '../constants/pages'
+
+// Components
+import { BoxUpload, BoxDownload, BoxNotAvailable } from '../components/box/Box'
+import Info from '../components/info/Info'
+
+class Page extends React.Component {
+  static propTypes = {
+    routeInfo: PropTypes.object.isRequired,
+    isLoading: PropTypes.bool.isRequired,
+    files: PropTypes.object.isRequired,
+    doFetchFileTree: PropTypes.func.isRequired,
+    doUpdateHash: PropTypes.func.isRequired,
+    currentPage: PropTypes.string.isRequired,
+    ipfsInitFailed: PropTypes.bool.isRequired,
+    shareLink: PropTypes.string
+  }
+
+  update (prevProps) {
```

Comment:
I think this can be refactored to:
```js
const { doUpdateHash, routeInfo: { params }, doFetchFileTree } = this.props

if (params.hash) {
  // Discard if it's not a valid hash
  if (!isIPFS.cid(params.hash)) {
	return doUpdateHash('#/')
  }

  doFetchFileTree(params.hash)
}
```

You can put it directly in the `componentDidMount` and remove the `componentDidUpdate`.